### ISSUE TITLE
fix(model): restore complete hybrid state on persistent prompt replacement

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12341]
Parent: #12235
Related: #9989

## Classification
- **Centrality:** Core
- **Worker tier:** Astra contract/review; Sol implementation
- **Write set:** `internal/model/qwen38_mtp_persistent.go`, `internal/model/qwen38_mtp_persistent_test.go`

## For / Problem / Today / Better because / Witness
- **For:** native Qwen3.8 callers that edit or shorten a persistent conversation.
- **Problem:** reused session state must equal a fresh session consuming exactly the replacement prompt.
- **Today:** `Prefill` calls `TryEvict` and ignores failure; hybrid recurrence eviction is rejected without changing the cache. Divergent and strict-prefix replacements retain discarded KV/recurrent state and can return stale logits.
- **Better because:** complete restoration or clean replay makes KV, recurrence, convolution windows, hidden/token history, and continuation logits match a fresh native session.
- **Witness:** `TestQwen38MTPPersistentPromptReplacementMatchesFresh`.

## Acceptance
- Add divergent `[0,1,2,3] -> [0,1,7]` and strict-prefix `[0,1,2,3] -> [0,1]` subcases.
- Compare complete state and one identical continuation step against independent fresh-session oracles.
- Preserve fak-native execution; no llama.cpp fallback.

## Witness command
`go test ./internal/model -run '^TestQwen38MTPPersistentPromptReplacementMatchesFresh$' -count=1`

## Non-goals
Prompt-cache policy redesign, wide-M verification, adaptive depth, and performance claims.

**Comment by anthony-chaudhary** [comment]
Claimed the assigned two-file correctness leaf. Invariant: replacement prefill must match an independent fresh native session in complete KV/lineage, recurrent state, convolution windows, hidden/token history, and logits, then match one identical continuation. Regression added; first execution was blocked before compilation by local disk exhaustion, so no passing witness is claimed.

**Comment by anthony-chaudhary** [comment]
The exact regression is now red-to-green: before the fix, divergent replacement left cache length 5 instead of 3 and strict-prefix replacement left 4 instead of 2. After clean native replay, both cases match independent fresh-session state and logits, including one identical continuation. Existing persistent-session tests and go vet ./internal/model also pass. Guarded landing is waiting on #12383: prospective cmd/fak vet reports the duplicate containsStr helper; validation also timed out. No commit or closure is claimed yet.

**Comment by anthony-chaudhary** [comment]
Software proof is green: the exact fresh-session regression passed after failing both required replacements before the fix; existing persistent-session tests and model vet passed. The exact regression also passed in the isolated prospective tree based on 726c52b64. Per coordinator adjudication this is a host-side correctness leaf; no hardware qualification is claimed, and the extra hardware classifier remains tracked by #12382. The instructed managed two-path landing, with required test witness, go-build verification and remote-recovery checks enabled, returned SYMPTOM_UNWITNESSED before mutation. Its symptom check resolves committed HEAD, while this isolated checkout has the verified uncommitted patch on a merge base. No unsafe skip flag was used. No commit or push was produced; the two-file patch is preserved and this issue remains open pending a sanctioned candidate-witness binding.

**Comment by anthony-chaudhary** [comment]
The prospective-candidate landing repair is shipped on `origin/main` in [`5623cb8df7c7`](https://github.com/anthony-chaudhary/fak/commit/5623cb8df7c701a3d334b50e89ab24afe716b077) (`cmd/fak r4070+g5623cb8df`, `internal/workerworktree r82+g5623cb8df`). The repaired path was dogfooded with exact-commit symptom verification before recovery publication/CAS.

The preserved two-file candidate can now retry without any unsafe skip. Build the shipped binary, then run:

```bash
GOBIN=/Users/anthony/.codex/worktrees/5833/fak/_scratch/issue-12341/bin \
  go install github.com/anthony-chaudhary/fak/cmd/fak@5623cb8df7c701a3d334b50e89ab24afe716b077

GOCACHE=/Users/anthony/.codex/worktrees/5833/fak/_scratch/issue-12341/gocache \
GOTMPDIR=/Users/anthony/.codex/worktrees/5833/fak/_scratch/issue-12341/gotmp \
GOMAXPROCS=2 GOFLAGS=-p=2 \
/Users/anthony/.codex/worktrees/5833/fak/_scratch/issue-12341/bin/fak worktree worker land \
  --root /Users/anthony/Documents/GitHub/fak \
  --worktree /Users/anthony/.codex/worktrees/5833/fak \
  --base-sha 7c23b70aab1af34d210604cb563f3601d6533a6f \
  --paths internal/model/qwen38_mtp_persistent.go \
  --paths internal/model/qwen38_mtp_persistent_test.go \
  --msg-file /Users/anthony/.codex/worktrees/5833/fak/_scratch/issue-12341/commit-message.txt \
  --verify go-build --require-test-witness \
  --recovery-remote origin --require-remote-recovery \
  --disambiguation-timeout-ms 120000 \
  > /Users/anthony/.codex/worktrees/5833/fak/_scratch/issue-12341/land.json
```

Preserved patch identity remains SHA-256 `1388b61fe87dcb688b38f2ac59679e0a21e60fa24b881dc060f328868a011111`. Do not add `--unsafe-skip-symptom-witness` or bypass the guarded lander.

**Comment by anthony-chaudhary** [comment]
Retry precondition: the selected `--root` must have `5623cb8df7c7` in its current `main` ancestry. Check with `git -C /Users/anthony/Documents/GitHub/fak merge-base --is-ancestor 5623cb8df7c701a3d334b50e89ab24afe716b077 HEAD` and `fak sync check --fetch --repo /Users/anthony/Documents/GitHub/fak`. The shared checkout is presently peer-dirty and its safe-sync route may refuse rather than overwrite that work; if the ancestry check is false, wait for/perform sanctioned `fak sync reconcile` convergence before running the retry. Do not reset, raw-merge, force-push, or retarget the candidate to bypass this precondition.
